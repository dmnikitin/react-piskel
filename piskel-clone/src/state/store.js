import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { getFromLocalStorage } from '../helpers/localStorageHandler';
import combined from './reducers/combined';

const localStorageData = getFromLocalStorage();
const store = createStore(combined, localStorageData, composeWithDevTools());

export default store;
