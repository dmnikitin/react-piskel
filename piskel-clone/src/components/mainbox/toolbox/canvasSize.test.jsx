import React from 'react'
import { shallow, mount } from 'enzyme';
import CanvasSize from './canvasSize';
import store from '../../../state/store';

const callback = () => { };

jest.mock('../button.jsx', () => 'Button');

const wrapperShallow = shallow(<CanvasSize callback={callback} />);
const wrapperMount = mount(<CanvasSize store={store} callback={callback} />);

describe('canvasSize component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it('displays initial div', () => {
    expect(wrapperMount.find("div")).toHaveLength(2);
  });
  it('displays proper amount of buttons', () => {
    expect(wrapperMount.find('.toolbox-canvas__buttons').children()).toHaveLength(3);
  });
  it('displays heading', () => {
    expect(wrapperMount.find('h3').text()).toBe('Choose the canvas size');
  });

});
