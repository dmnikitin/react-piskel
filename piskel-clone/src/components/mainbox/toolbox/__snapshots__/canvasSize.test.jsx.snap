// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`canvasSize component renders correctly 1`] = `
<div
  className="toolbox-canvas"
>
  <h3>
    Choose the canvas size
  </h3>
  <div
    className="toolbox-canvas__buttons"
  >
    <Button
      callback={[Function]}
      data="32"
      icon="square-o"
    />
    <Button
      callback={[Function]}
      data="64"
      icon="plus-square-o"
    />
    <Button
      callback={[Function]}
      data="128"
      icon="plus-square"
    />
  </div>
</div>
`;
